src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -259,6 +259,11 @@ public QuorumServer(long sid, String addressStr) throws ConfigException {
                 throw new ConfigException("Address unresolved: " + serverParts[0] + ":" + serverParts[2]);
             }
 
+            if(addr.getPort() == electionAddr.getPort()) {
+                throw new ConfigException(
+                        "Client and election port must be different! Please update the configuration file on server." + sid);
+            }
+
```

Comment:
Does it make sense to apply a similar change to the 
`public QuorumServer(long id, InetSocketAddress addr,
InetSocketAddress electionAddr, InetSocketAddress clientAddr, LearnerType type) {`
constructor below?

Reply:
Good question. Well, as what I can see it is only used for test, and it is called programmatically, so not from reading a config. The intention here was to make sure that ZK is not misconfigured. 

So I'm open to adding it there also, just my initial thinking was it is not necessary.

Reply:
Double checked and you're right, all the other constructors for this class are only invoked in tests. Agreed that my proposal is a "nice to have" at best.